app/containers/App/index.js

```diff
@@ -8,38 +8,61 @@
 
 import React from 'react';
 import Helmet from 'react-helmet';
+import { withRouter } from 'react-router';
 
 // Import the CSS reset, which HtmlWebpackPlugin transfers to the build folder
 import 'sanitize.css/sanitize.css';
 
+import ProgressBar from 'components/ProgressBar';
 import Img from 'components/Img';
 import Footer from 'components/Footer';
 import Banner from './banner-metal.jpg';
 import A from 'components/A';
 
 import styles from './styles.css';
 
-function App(props) {
-  return (
-    <div className={styles.wrapper}>
-      <Helmet
-        titleTemplate="%s - React.js Boilerplate"
-        defaultTitle="React.js Boilerplate"
-        meta={[
-          { name: 'description', content: 'A React.js Boilerplate application' },
-        ]}
-      />
-      <A className={styles.logoWrapper} href="https://twitter.com/mxstbr">
-        <Img className={styles.logo} src={Banner} alt="react-boilerplate - Logo" />
-      </A>
-      {React.Children.toArray(props.children)}
-      <Footer />
-    </div>
-  );
+export class App extends React.Component {
+
+  state = {
+    progress: -1,
+  };
+
+  componentDidMount() {
+    this.props.router.listen(() => this.setState({ progress: 0 }));
```

Comment:
nice work 👍

one quick thing... we need to store a reference to listener...

```js
this.unsubscribeHistory = this.props.router.listen(() => this.setState({ progress: 0 }));
```

and use lifecycle hook for teardown

```js
//...
componentWillUnmount () {
  this.unsubscribeHistory
}
```

otherwise this will cause memory leak since listeners won't be garbage collected :)